contents/tutorials/flutter-feature-flags.md

```diff
@@ -0,0 +1,289 @@
+import CreateFlagDark from '../images/tutorials/flutter-feature-flags/create-flag-dark.png'
+
+[Feature flags](/feature-flags) help you conditionally roll out and release features safely. This tutorial shows you how integrate them into your Flutter app using PostHog. 
+
+We'll create a basic Flutter app, add PostHog, create a feature flag, and then implement the flag to control content in our app.
+
+## 1. Create a new Flutter app
+
+Our app will have two screens:
+
+1. The first screen will have a button which takes you to a second screen.
+2. The second screen will either have a `red` or `green` background color, depending on whether our feature flag is enabled or not.
+
+To set this up, first ensure the [Flutter extension for VS Code](https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter) is installed. Then, create a new app by opening the Command Palette in VS Code (`Ctrl/Cmd + Shift + P`), typing `flutter` and selecting `Flutter: New Project`. 
+
+Select `Empty Application` and name your app `flutter_feature_flags`. Then, replace your code in `lib/main.dart` with the following:
+
+```dart file=lib/main.dart
+import 'package:flutter/material.dart';
+import 'feature_screen_view.dart';
+
+void main() {
+  runApp(const MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      title: 'Flutter Feature Flags App',
+      home: MainScreen(),
+    );
+  }
+}
+
+class MainScreen extends StatelessWidget {
+  const MainScreen({super.key});
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Main Screen')),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            const Text('Hello, world!'),
+            ElevatedButton(
+              child: const Text('Go to Next Screen'),
+              onPressed: () {
+                Navigator.push(
+                  context,
+                  MaterialPageRoute(builder: (context) => const FeatureScreenView(isFlagEnabled: false)), // We update this later
+                );
+              },
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+```
+
+Lastly, in the `lib` directory, create a new file for our second screen called `feature_screen_view.dart`. Add the following code to it:
+
+```dart
+import 'package:flutter/material.dart';
+
+class FeatureScreenView extends StatelessWidget {
+  final bool isFlagEnabled;
+
+  const FeatureScreenView({Key? key, required this.isFlagEnabled}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: isFlagEnabled ? Colors.green : Colors.red,
+    );
+  }
+}
+```
+
+Press **F5** and run the app in any emulator (we chose Android) to see your app in action.
```

Comment:
They might have to have Android Studio running to launch the emulator. I launched with CMD + SHIFT + P, `Flutter: Select Device`.

Reply:
Lol I didnt need to have android studio installed, I think Flutter is just finicky like this. I think its okay to exclude, given that flutter devs will know how to do this